Streetcode/Streetcode.XUnitTest/MediatRTests/Media/Image/CreateImage/CreateImageTests.cs

```diff
@@ -0,0 +1,182 @@
+
+            SetupCreateRepository(1);
+            SetupBlobService();
+            SetupMapper(testImage, testImageDTO);
+
+            var handler = new CreateImageHandler(_mockBlobService.Object, _mockRepository.Object, _mockLogger.Object, _mockMapper.Object, _mockLocalizerFail.Object, _mockLocalizerConvertNull.Object);
+
+            // Act
+            var result = await handler.Handle(new CreateImageCommand(testCreateImageDTO), CancellationToken.None);
+
+            // Assert
+            Assert.True(result.IsSuccess);
+        }
+
+        [Fact]
+        public async Task ShouldThrowException_AltIsRequired()
+        {
+            // Arrange
+            var expectedError = "Cannot create image without alt";
+            _mockLocalizerConvertNull.Setup(x => x["CannotCreateImageWithoutAlt"])
+            .Returns(new LocalizedString("CannotConvertNullToCategory", expectedError));
+
+            var testCreateImageDTO = GetImageCreateDTOWithoutAlt();
+            var testImageDTO = GetImageDTO();
+            var testImage = GetImage();
+
+            SetupCreateRepository(1);
+            SetupBlobService();
+            SetupMapper(testImage, testImageDTO);
+
+            var handler = new CreateImageHandler(_mockBlobService.Object, _mockRepository.Object, _mockLogger.Object, _mockMapper.Object, _mockLocalizerFail.Object, _mockLocalizerConvertNull.Object);
+
+            // Act
+            var result = await handler.Handle(new CreateImageCommand(testCreateImageDTO), CancellationToken.None);
+
+            // Assert
+            Assert.False(result.IsSuccess);
+            Assert.Equal(expectedError, result.Errors.First().Message);
+        }
+
+        [Fact]
+        public async Task ShouldThrowException_SaveChangesAsyncIsNotSuccessful()
+        {
+            // Arrange
+            var expectedError = "Failed to create an image";
+            _mockLocalizerFail.Setup(x => x["FailedToCreateAnImage"])
+            .Returns(new LocalizedString("FailedToCreateAnImage", expectedError));
+
+            var testCreateImageDTO = GetCreateImageDTO();
+            var testImageDTO = GetImageDTO();
+            var testImage = GetImage();
+
+            SetupCreateRepository(-1);
+            SetupBlobService();
+            SetupMapper(testImage, testImageDTO);
+
+            var handler = new CreateImageHandler(_mockBlobService.Object, _mockRepository.Object, _mockLogger.Object, _mockMapper.Object, _mockLocalizerFail.Object, _mockLocalizerConvertNull.Object);
+
+            // Act
+            var result = await handler.Handle(new CreateImageCommand(testCreateImageDTO), CancellationToken.None);
+
+            // Assert
+            Assert.True(result.IsFailed);
+            Assert.Equal(expectedError, result.Errors.First().Message);
+        }
+
+        private void SetupCreateRepository(int returnNumber)
+        {
+            _mockRepository.Setup(x => x.ImageRepository.Create(It.IsAny<DAL.Entities.Media.Images.Image>()));
```

Comment:
import ImageEntity at the top of the file like this: using ImageEntity = DAL.Entities.Media.Images.Image
And then use ImageEntity when needed